expression/builtin_string_serial_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"testing"
+	"unicode/utf8"
+
+	"github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/collate"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCIWeightString(t *testing.T) {
```

Comment:
Shall the `TestWeightString` should be moved to here too? it seems that https://github.com/pingcap/tidb/issues/29283 happens on `TestWeightString`

Reply:
Nope. The root cause of `TestWeightString` is running concurrently with `TestCIWeightString`. Thus after we move `TestCIWeightString` out, `TestWeightString` will be fixed.